# File list

### Scripts:

1. Compute read remapping rates on assembled genes (proxy for gene expression): remap_expr.sh

2. Functionnal annotations of predicted proteins: eggNOG_fun.sh

3. Graphical visualisation: scRNA_paper_Acanth.Rmd, scRNA_paper_Acanth_suppl.Rmd (cf. homepage)


### Single-cell transcriptomes and predicted proteomes:

Input of step 1.: transcriptomes (cf. Data)

Input of step 2.: proteomes (cf. Data)


### Dataframes:

* Fig3C: NMDS based on COG differential expression

* Fig3D: heatmap of COG expression in reproductive stages compared to vegetative

* FigSX: NMDS only Acantharia
